phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java

```diff
@@ -657,6 +660,22 @@ public static PName padTenantIdIfNecessary(RowKeySchema schema, boolean isSalted
         return tenantId;
     }
 
+    public static byte[] getTenantIdBytes(RowKeySchema schema, boolean isSalted, PName tenantId)
+            throws SQLException {
+        int pkPos = isSalted ? 1 : 0;
+        Field field = schema.getField(pkPos);
+        PDataType dataType = field.getDataType();
+        byte[] convertedValue;
+        try {
+            Object value = dataType.toObject(tenantId.getString());
```

Comment:
For this dataType.toBytes() call, use the one that accepts sortOrder so that it'll still work if the tenantId column is descending:

    convertedValue = dataType.toBytes(value, field.getSortOrder());